### PR TITLE
NIAD-24: enhancement produce a postman collection of requests that can be made to adaptors

### PR DESCRIPTION
**Pull request by Alicja-Kainos**

### CODE REVIEW

File: rest-client/mhs/synchronous-messaging-pattern/sync_soap_error.http [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Message ID: F5187FB6-B033-4A75-838B-9E7A1AFB3111 configured in fakespine to return a SOAP Fault error.
```
Comment (Alicja-Kainos) [inline rest-client/mhs/synchronous-messaging-pattern/sync_soap_error.http]: @Broom94, why did you add this file?

**Review by jaroslaw-wojciechowski — CHANGES_REQUESTED** [review]
question about 
rest-client/mhs/inbound/Solicited-request.http
rest-client/mhs/inbound/Unsolicited-request.http

both over 13k lines - is it necessary?

**Comment by Alicja-Kainos** [comment]
> question about
> rest-client/mhs/inbound/Solicited-request.http
> rest-client/mhs/inbound/Unsolicited-request.http
> 
> both over 13k lines - is it necessary?

**Comment by Alicja-Kainos** [comment]
> question about
> rest-client/mhs/inbound/Solicited-request.http
> rest-client/mhs/inbound/Unsolicited-request.http
> 
> both over 13k lines - is it necessary?

@Broom94 , why is it all in base64 and why this files are so huge? Is it necessary?

File: .gitignore [hunk 2]
```diff
@@ -147,8 +147,5 @@ export-env-vars-and-run-mhs-docker.sh
 # system
 .DS_Store
 
-# enviornment file
-integration-tests/integration_tests/integration-tests-env.yaml
```
Comment (bartek-sarul) [inline .gitignore:151]: In previous PR you removed it from `# Local configuration` and added here. I wrote to leave it there in `# Local configuration`. Here I see you are rolling back half of your change

File: rest-client/mhs/inbound/Solicited-request.http [hunk 3]
```diff
@@ -0,0 +1,13252 @@
+# In inbound main.py, change ssl_ctx.verify_mode = ssl.CERT_REQUIRED to ssl.CERT_NONE or ssl.CERT_OPTIONAL
+# This request will be throw 500 server error because of RefToMessageId. This is the refers to message id
+# to the 'original' message from outbound. Add this id to state database and specific workflow. 
+
+# @name Solicited-request
+POST {{HOSTNAME}}:{{INBOUND-PORT}}/ HTTP/1.1
+Host: rmmid.nis1.national.ncrs.nhs.uk
+SOAPAction: urn:nhs:names:services:gp2gp/COPC_IN000001UK01
+Content-Length: 1028570
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/Solicited-request.http]: No! Please do not commit such large files!
This is an example request that can be much smaller!
Look at the attachments. The b64 object i soooo large. As an example you can use something much smaller like "Hello, World!" encoded

File: rest-client/mhs/spine/async_express.http [hunk 4]
```diff
@@ -1,6 +1,6 @@
 
 # @name Async_Express
-POST https://192.168.128.11/reliablemessaging/queryrequest HTTP/1.1
+POST {{SPINE_HOSTNAME}}/reliablemessaging/queryrequest HTTP/1.1
```
Comment (bartek-sarul) [inline rest-client/mhs/spine/async_express.http]: hot hostname but "base_url"
Hostname would be 192.168.128.11 only (kind of)
Here you expect domain and protocol

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (bartek-sarul) [reply .gitignore:151]: ahh sorry, I didn't see it's there

**Comment by Broom94** [comment]
> ver 13k lines - is it ne

You can remove attachment, could be useful at sometime though. So you could make it a lot smaller, look at Bartosz comment.

File: rest-client/mhs/inbound/Solicited-request.http (hunk 3 above)
Comment (Alicja-Kainos) [reply rest-client/mhs/inbound/Solicited-request.http]: @bartek-sarul , @Broom94 ,  I've just replaced it with "Hello World"

File: rest-client/mhs/inbound/Solicited-request.http [hunk 5]
```diff
@@ -0,0 +1,130 @@
+# In inbound main.py, change ssl_ctx.verify_mode = ssl.CERT_REQUIRED to ssl.CERT_NONE or ssl.CERT_OPTIONAL
+# This request will be throw 500 server error because of RefToMessageId. This is the refers to message id
+# to the 'original' message from outbound. Add this id to state database and specific workflow. 
+
+# @name Solicited-request
+
+
+@baseUrl = http://localhost:
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/Solicited-request.http]: I guess base url shouldn't have colon in it

File: rest-client/mhs/outbound/Async-Express-Pattern-Message.http [hunk 6]
```diff
@@ -1,10 +1,10 @@
 # @name Async_Express_Pattern_Message
 
 POST {{HOSTNAME}}:{{OUTBOUND-PORT}} HTTP/1.1
```
Comment (bartek-sarul) [inline rest-client/mhs/outbound/Async-Express-Pattern-Message.http]: here as well I think it should be more like "base url" and not hostname as it also included protocol

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: rest-client/mhs/outbound/Forward-Reliable-Pattern-Message-Asynchronous-Response.http [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # @name Forward_Reliable_Pattern_Message_Asynchronous_Response
```
Comment (bartek-sarul) [inline rest-client/mhs/outbound/Forward-Reliable-Pattern-Message-Asynchronous-Response.http]: Can you check the payload does it have configurable party from and party to fields? Those define where this message should be forwarded to after full circle

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

**Review by michael-kainos — CHANGES_REQUESTED** [review]
We seem to be missing a success request in these two message patterns:

asynchronous-express-message-pattern
synchronous-messaging-pattern

**Comment by Alicja-Kainos** [comment]
> We seem to be missing a success request in these two message patterns:
> 
> asynchronous-express-message-pattern
> synchronous-messaging-pattern

rest-client/mhs/outbound/Async-Express-Pattern-Message.http
rest-client/mhs/outbound/Synchronous-Pattern-Message.http 
@michael-kainos , are that these request?

**Comment by michael-kainos** [comment]
> > We seem to be missing a success request in these two message patterns:
> > asynchronous-express-message-pattern
> > synchronous-messaging-pattern
> 
> rest-client/mhs/outbound/Async-Express-Pattern-Message.http
> rest-client/mhs/outbound/Synchronous-Pattern-Message.http
> @michael-kainos , are that these request?

Not sure if they are success requests, please can you confirm? If these are success, please can you move to the correct directory for a message pattern and rename to match the format of the other requests.

**Comment by Alicja-Kainos** [comment]
> > > We seem to be missing a success request in these two message patterns:
> > > asynchronous-express-message-pattern
> > > synchronous-messaging-pattern
> > 
> > 
> > rest-client/mhs/outbound/Async-Express-Pattern-Message.http
> > rest-client/mhs/outbound/Synchronous-Pattern-Message.http
> > @michael-kainos , are that these request?
> 
> Not sure if they are success requests, please can you confirm? If these are success, please can you move to the correct directory for a message pattern and rename to match the format of the other requests.

Synchronous is successful for sure. What about the async express, I remember it was successful, today gives me 500, I'll se why. What do you mean saying the correct director, @michael-kainos ? This are requests for outbound so are placed in the outbound directory. And what about this pattern for file name - are there any rules? I think these names are clear.

**Comment by Alicja-Kainos** [comment]
> > > We seem to be missing a success request in these two message patterns:
> > > asynchronous-express-message-pattern
> > > synchronous-messaging-pattern
> > 
> > 
> > rest-client/mhs/outbound/Async-Express-Pattern-Message.http
> > rest-client/mhs/outbound/Synchronous-Pattern-Message.http
> > @michael-kainos , are that these request?
> 
> Not sure if they are success requests, please can you confirm? If these are success, please can you move to the correct directory for a message pattern and rename to match the format of the other requests.

Yes, all the request for outbound are successful. I have just double checked locally,

File: rest-client/README.md [hunk 8]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+{
+    "python.pythonPath": "/usr/local/bin/python3",
+    "workbench.settings.editor": "json",
+    "workbench.settings.useSplitJSON": true,
+    "rest-client.environmentVariables": {
+        "$shared": {},
+        "$sample_mhs_environment": {
+            "BASE_URL": "http://localhost",
+            "INBOUND-PORT": "443",
+            "TO-PARTY-ID": "X26-9199246",
```
Comment (bartek-sarul) [inline rest-client/README.md]: Isn't `to-party-id` const but the `from-party-id` variable?

File: rest-client/README.md [hunk 9]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+            "OUTBOUND-PORT": "80",
+            "ROUTE-LOOKUP-PORT": "8088",
+            "FAKE-SPINE-PORT": "8091",
+            "ASID": "9XXXXXXXXXXX",
+            "GUID-idRoot": "3c996a6b-8e3b-42ec-8796-34aa8dca353c",
```
Comment (bartek-sarul) [inline rest-client/README.md]: Can you add some description where and why is it a variable and not generated on the fly?

File: rest-client/README.md [hunk 10]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+            "GUID-idAltRoot": "af16f8c8-1a31-470c-802a-1716425b04fa",
```
Comment (bartek-sarul) [inline rest-client/README.md]: Can you add some description where and why is it a variable and not generated on the fly?

File: rest-client/README.md [hunk 11]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+            "SHORT-TIMESTAMP": "20190927152035",
+            "PARTY-KEY": "A9XXXX-XXXXXXX",
```
Comment (bartek-sarul) [inline rest-client/README.md]: What's the difference between `party-key` and `to/from party id`?

File: rest-client/README.md [hunk 12]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+            "FILE-UPLOADED": "test file will go here",
```
Comment (bartek-sarul) [inline rest-client/README.md]: Should this be a variable?

File: rest-client/README.md [hunk 13]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+            "LOCAL_SCR_URL": "http://localhost:9000/",
```
Comment (bartek-sarul) [inline rest-client/README.md]: Can be `SCR_URL`

File: rest-client/README.md [hunk 14]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+            "MESSAGE_ID": "6a6c8179-e8e3-419a-a6dc-0c07844ddef7"
```
Comment (bartek-sarul) [inline rest-client/README.md]: Should not be a variable but autogenerated on the fly when request is made {{$GUID}}

File: rest-client/README.md [hunk 15]
```diff
@@ -115,19 +33,46 @@ Specify file path to use as a body:
...
+            "BASE_URL": "http://localhost",
+            "INBOUND-PORT": "443",
+            "TO-PARTY-ID": "X26-9199246",
+            "OUTBOUND-PORT": "80",
+            "ROUTE-LOOKUP-PORT": "8088",
+            "FAKE-SPINE-PORT": "8091",
+            "ASID": "9XXXXXXXXXXX",
+            "GUID-idRoot": "3c996a6b-8e3b-42ec-8796-34aa8dca353c",
+            "GUID-idAltRoot": "af16f8c8-1a31-470c-802a-1716425b04fa",
+            "SHORT-TIMESTAMP": "20190927152035",
```
Comment (bartek-sarul) [inline rest-client/README.md]: Does this have to be a variable?

File: rest-client/mhs/inbound/solicited-request.http [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+
+# @name solicited-request
+
+
+@baseUrl = http://localhost
+@port = 443
+ 
+# @name Request
+POST {{baseUrl}}:{{port}}/ HTTP/1.1
+Host: rmmid.nis1.national.ncrs.nhs.uk
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/solicited-request.http]: Not needed

File: rest-client/mhs/inbound/solicited-request.http [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+Content-Id: <ebXMLHeader@spine.nhs.uk>
+Content-Type: text/xml; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+<?xml version="1.0" encoding="UTF-8"?>
+<SOAP:Envelope xmlns:xsi="http://www.w3c.org/2001/XML-Schema-Instance" xmlns:SOAP="http://schemas.xmlsoap.org/soap/envelope/" xmlns:eb="http://www.oasis-open.org/committees/ebxml-msg/schema/msg-header-2_0.xsd" xmlns:hl7ebxml="urn:hl7-org:transport/ebxml/DSTUv1.0" xmlns:xlink="http://www.w3.org/1999/xlink">
+<SOAP:Header>
+	<eb:MessageHeader SOAP:mustUnderstand="1" eb:version="2.0">
+		<eb:From>
+			<eb:PartyId eb:type="urn:nhs:names:partyType:ocs+serviceInstance">RHM-801710</eb:PartyId>
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/solicited-request.http]: Keys in From and To should be variable

File: rest-client/mhs/inbound/solicited-request.http [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+		</eb:From>
+		<eb:To>
+			<eb:PartyId eb:type="urn:nhs:names:partyType:ocs+serviceInstance">A91576-9198964</eb:PartyId>
+		</eb:To>
+		<eb:CPAId>S2030731A2137927</eb:CPAId>
+		<eb:ConversationId>74799658-43DF-11E3-805D-BBA66E7A9031</eb:ConversationId>
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/solicited-request.http]: Conversation ID can be the same as message id or just new GUID

File: rest-client/mhs/inbound/solicited-request.http [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+		<eb:Service>urn:nhs:names:services:gp2gp</eb:Service>
+		<eb:Action>COPC_IN000001UK01</eb:Action>
+		<eb:MessageData>
+			<eb:MessageId>047C22B4-613F-47D3-9A72-44A1758464FB</eb:MessageId>
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/solicited-request.http]: MessageId shuold be a new random guid

File: rest-client/mhs/inbound/solicited-request.http [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+			<eb:Timestamp>2013-11-04T14:33:58Z</eb:Timestamp>
+			<eb:RefToMessageId>8936cd14-d728-41a7-8d14-485183ebc2de</eb:RefToMessageId>
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/solicited-request.http]: This has to be set by the user so it should be a variable

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

**Comment by michael-kainos** [comment]
> > > > We seem to be missing a success request in these two message patterns:
> > > > asynchronous-express-message-pattern
> > > > synchronous-messaging-pattern
> > > 
> > > 
> > > rest-client/mhs/outbound/Async-Express-Pattern-Message.http
> > > rest-client/mhs/outbound/Synchronous-Pattern-Message.http
> > > @michael-kainos , are that these request?
> > 
> > 
> > Not sure if they are success requests, please can you confirm? If these are success, please can you move to the correct directory for a message pattern and rename to match the format of the other requests.
> 
> Synchronous is successful for sure. What about the async express, I remember it was successful, today gives me 500, I'll se why. What do you mean saying the correct director, @michael-kainos ? This are requests for outbound so are placed in the outbound directory. And what about this pattern for file name - are there any rules? I think these names are clear.

It seems the directories for outbound, inbound etc.. are one migrated over from postman. We should only require the requests in the directories that are named after the message pattern. Please remove the other older requests, e.g. outbound, inbound, spine.

It did not work for me locally so it seems there is some local config needed.  May be worth adding more instructions in a readme file.

Also, i have committed some changes to 2 requests i am using in a demo tomorrow for the supplier. The change you will see is the message-id being randomly generated. We should probably update this for all requests.

**Comment by Alicja-Kainos** [comment]
> > > > > We seem to be missing a success request in these two message patterns:
> > > > > asynchronous-express-message-pattern
> > > > > synchronous-messaging-pattern
> > > > 
> > > > 
> > > > rest-client/mhs/outbound/Async-Express-Pattern-Message.http
> > > > rest-client/mhs/outbound/Synchronous-Pattern-Message.http
> > > > @michael-kainos , are that these request?
> > > 
> > > 
> > > Not sure if they are success requests, please can you confirm? If these are success, please can you move to the correct directory for a message pattern and rename to match the format of the other requests.
> > 
> > 
> > Synchronous is successful for sure. What about the async express, I remember it was successful, today gives me 500, I'll se why. What do you mean saying the correct director, @michael-kainos ? This are requests for outbound so are placed in the outbound directory. And what about this pattern for file name - are there any rules? I think these names are clear.
> 
> It seems the directories for outbound, inbound etc.. are one migrated over from postman. We should only require the requests in the directories that are named after the message pattern. Please remove the other older requests, e.g. outbound, inbound, spine.
> 
> It did not work for me locally so it seems there is some local config needed. May be worth adding more instructions in a readme file.
> 
> Also, i have committed some changes to 2 requests i am using in a demo tomorrow for the supplier. The change you will see is the message-id being randomly generated. We should probably update this for all requests.

Local configuration is only needed for inbound, what is written in the comment in the two inboud request files. You need to switch off ssl. For other, just running inbound, outbound and spineroutelookup services is enough.

**Comment by michael-kainos** [comment]
> > > > > > We seem to be missing a success request in these two message patterns:
> > > > > > asynchronous-express-message-pattern
> > > > > > synchronous-messaging-pattern
> > > > > 
> > > > > 
> > > > > rest-client/mhs/outbound/Async-Express-Pattern-Message.http
> > > > > rest-client/mhs/outbound/Synchronous-Pattern-Message.http
> > > > > @michael-kainos , are that these request?
> > > > 
> > > > 
> > > > Not sure if they are success requests, please can you confirm? If these are success, please can you move to the correct directory for a message pattern and rename to match the format of the other requests.
> > > 
> > > 
> > > Synchronous is successful for sure. What about the async express, I remember it was successful, today gives me 500, I'll se why. What do you mean saying the correct director, @michael-kainos ? This are requests for outbound so are placed in the outbound directory. And what about this pattern for file name - are there any rules? I think these names are clear.
> > 
> > 
> > It seems the directories for outbound, inbound etc.. are one migrated over from postman. We should only require the requests in the directories that are named after the message pattern. Please remove the other older requests, e.g. outbound, inbound, spine.
> > It did not work for me locally so it seems there is some local config needed. May be worth adding more instructions in a readme file.
> > Also, i have committed some changes to 2 requests i am using in a demo tomorrow for the supplier. The change you will see is the message-id being randomly generated. We should probably update this for all requests.
> 
> Local configuration is only needed for inbound, what is written in the comment in the two inboud request files. You need to switch off ssl. For other, just running inbound, outbound and spineroutelookup services is enough.

Ignore this comment from me, lets speak with the guys in tomorrows stand up for the structure of requests and any that should be removed

**Review by michael-kainos — APPROVED** [review]

**Review by Broom94 — APPROVED** [review]

**Review by michael-kainos — APPROVED** [review]
